ui/src/main/java/com/xmartlabs/bigbang/ui/common/parcel/ParcelerEnumTypeConverter.kt

```diff
@@ -0,0 +1,21 @@
+package com.xmartlabs.bigbang.ui.common.parcel
+
+import org.parceler.ParcelConverter
+import java.util.*
+
+@Suppress("unused")
+open class ParcelerEnumTypeConverter<T : Enum<T>>(val clazz: Class<T>) : ParcelConverter<T> {
+  override fun toParcel(input: T?, parcel: android.os.Parcel) {
+    parcel.writeInt(input?.ordinal ?: -1)
+  }
+
+  override fun fromParcel(parcel: android.os.Parcel): T? {
+    val ordinal = parcel.readInt()
+    if (ordinal == -1) {
```

Comment:
The ordinal can never be -1. And Parceler plays a lot with -1 as the "null" value (see https://github.com/johncarl81/parceler/blob/92c2d517a9a28dca018c2eb16dac640c8bbde3de/parceler-api/src/main/java/org/parceler/converter/CollectionParcelConverter.java#L29).

Btw, maybe @matir91 you should use a `NULL` constant too.